packages/block-editor/src/hooks/utils.js

```diff
@@ -126,48 +126,93 @@ export function shouldSkipSerialization( blockType, featureSet, feature ) {
-	const textColumns = useSetting( 'typography.textColumns' );
-	const textDecoration = useSetting( 'typography.textDecoration' );
-	const writingMode = useSetting( 'typography.writingMode' );
-	const textTransform = useSetting( 'typography.textTransform' );
-	const letterSpacing = useSetting( 'typography.letterSpacing' );
-	const padding = useSetting( 'spacing.padding' );
-	const margin = useSetting( 'spacing.margin' );
-	const blockGap = useSetting( 'spacing.blockGap' );
-	const spacingSizes = useSetting( 'spacing.spacingSizes' );
-	const units = useSetting( 'spacing.units' );
-	const minHeight = useSetting( 'dimensions.minHeight' );
-	const layout = useSetting( 'layout' );
-	const borderColor = useSetting( 'border.color' );
-	const borderRadius = useSetting( 'border.radius' );
-	const borderStyle = useSetting( 'border.style' );
-	const borderWidth = useSetting( 'border.width' );
-	const customColorsEnabled = useSetting( 'color.custom' );
-	const customColors = useSetting( 'color.palette.custom' );
-	const customDuotone = useSetting( 'color.customDuotone' );
-	const themeColors = useSetting( 'color.palette.theme' );
-	const defaultColors = useSetting( 'color.palette.default' );
-	const defaultPalette = useSetting( 'color.defaultPalette' );
-	const defaultDuotone = useSetting( 'color.defaultDuotone' );
-	const userDuotonePalette = useSetting( 'color.duotone.custom' );
-	const themeDuotonePalette = useSetting( 'color.duotone.theme' );
-	const defaultDuotonePalette = useSetting( 'color.duotone.default' );
-	const userGradientPalette = useSetting( 'color.gradients.custom' );
-	const themeGradientPalette = useSetting( 'color.gradients.theme' );
-	const defaultGradientPalette = useSetting( 'color.gradients.default' );
-	const defaultGradients = useSetting( 'color.defaultGradients' );
-	const areCustomGradientsEnabled = useSetting( 'color.customGradient' );
-	const isBackgroundEnabled = useSetting( 'color.background' );
-	const isLinkEnabled = useSetting( 'color.link' );
-	const isTextEnabled = useSetting( 'color.text' );
-	const isHeadingEnabled = useSetting( 'color.heading' );
-	const isButtonEnabled = useSetting( 'color.button' );
+	const [
+		fontFamilies,
+		fontSizes,
+		customFontSize,
+		fontStyle,
+		fontWeight,
+		lineHeight,
+		textColumns,
+		textDecoration,
+		writingMode,
+		textTransform,
+		letterSpacing,
+		padding,
+		margin,
+		blockGap,
+		spacingSizes,
+		units,
+		minHeight,
+		layout,
+		borderColor,
+		borderRadius,
+		borderStyle,
+		borderWidth,
+		customColorsEnabled,
+		customColors,
+		customDuotone,
+		themeColors,
+		defaultColors,
+		defaultPalette,
+		defaultDuotone,
+		userDuotonePalette,
+		themeDuotonePalette,
+		defaultDuotonePalette,
+		userGradientPalette,
+		themeGradientPalette,
+		defaultGradientPalette,
+		defaultGradients,
+		areCustomGradientsEnabled,
+		isBackgroundEnabled,
+		isLinkEnabled,
+		isTextEnabled,
+		isHeadingEnabled,
+		isButtonEnabled,
+	] = useSettings( [
```

Comment:
I think it could be nice to consider building an ESLint rule that prevents folks from having multiple `useSetting` calls at the same level, recommending that we use `useSettings()` instead.

Not for this PR, obviously, but something that we discussed earlier and I wanted to make sure it doesn't fall through the cracks.